superduper/backends/base/query.py

```diff
@@ -456,17 +529,25 @@ def _encode_or_unpack_args(self, r, db, method='encode', parent=None):
         return r
 
     def _execute(self, parent, method='encode'):
-        for part in self.parts:
-            if isinstance(part, str):
-                parent = getattr(parent, part)
-                continue
-            args = self._encode_or_unpack_args(
-                part[1], self.db, method=method, parent=parent
-            )
-            kwargs = self._encode_or_unpack_args(
-                part[2], self.db, method=method, parent=parent
-            )
-            parent = getattr(parent, part[0])(*args, **kwargs)
+        return self._get_chain_native_query(parent, self.parts, method)
+
+    def _get_chain_native_query(self, parent, parts, method='encode'):
+        try:
+            for part in parts:
+                if isinstance(part, str):
+                    parent = getattr(parent, part)
+                    continue
+                args = self._encode_or_unpack_args(
+                    part[1], self.db, method=method, parent=parent
+                )
+                kwargs = self._encode_or_unpack_args(
+                    part[2], self.db, method=method, parent=parent
+                )
+                parent = getattr(parent, part[0])(*args, **kwargs)
+        except Exception as e:
+            logging.error(f'Error in executing query, parts: {parts}')
+            raise e
+
```

Comment:
Separating this method allows for easier future expansion of the query, enabling the query to flexibly construct the parts list without being tied to the expression. This way, all Queries can be merged into one, with Query only recording the expression and being independent of the database.

Reply:
Ok, yes this is a possible approach.